app/models/miq_provision_orch_workflow.rb

```diff
@@ -0,0 +1,66 @@
+class MiqProvisionOrchWorkflow < MiqProvisionVirtWorkflow
+  def initialize(values, requester, options = {})
+    initial_pass = values.blank?
+    initial_pass = true if options[:initial_pass] == true
+    instance_var_init(values, requester, options)
+
+    # Check if the caller passed the source VM as part of the initial call
+    if initial_pass == true
+      src_vm_id = get_value(@values[:src_vm_id])
+      unless src_vm_id.blank?
+        vm = OrchestrationTemplate.find_by(:id => src_vm_id)
+        @values[:src_vm_id] = [vm.id, vm.name] unless vm.blank?
+      end
+    end
+
+    unless options[:skip_dialog_load] == true
+      # If this is the first time we are called the values hash will be empty
+      # Also skip if we are being called from a web-service
+      @dialogs = get_pre_dialogs if initial_pass && options[:use_pre_dialog] != false
+      if @dialogs.nil?
+        @dialogs = get_dialogs
+      else
+        @running_pre_dialog = true if options[:use_pre_dialog] != false
+      end
+      normalize_numeric_fields unless @dialogs.nil?
+    end
+
+    password_helper(@values, false) # Decrypt passwords in the hash for the UI
+    @last_vm_id = get_value(@values[:src_vm_id]) unless initial_pass == true
+
+    return if options[:skip_dialog_load] == true
+
+    set_default_values
+    update_field_visibility
+
+    if get_value(values[:service_template_request])
+      show_dialog(:requester, :hide, "disabled")
+      show_dialog(:purpose,   :hide, "disabled")
+    end
+  end
+
+  def get_source_and_targets(refresh = false)
```

Comment:
This method is copied from `MiqProvisionVirtWorkflow` too.
Used `OrchestrationTemplate` as the template class and deleted `@vm_snapshot_count`.